### DISCUSSION TITLE
Testes End-to-End

### DISCUSSION THREAD

**Discussion by toninlopes** [diego3g/rsxp-2023#113] [Q&A] [answered]
A ideia aqui é discutir a inclusão de testes end-to-end. Por exemplo, podemos adicionar no projeto o framework [Detox](https://wix.github.io/Detox/). Segue aqui alguns possíveis benefícios em adicionar testes end-to-end:

- Testar as funcionalidades como se fosse um usuário comum;
- Possibilidade de rodar os testes end-to-end nas PRs para garantir que as funcionalidades não quebrem;
- Possibilidade de reduzir o tempo em teste de regressão, ja que parte dos testes serão automatizados.

**Comment by Rodrigo001-dev** [comment, marked as answer]
@aclopesjr Opa blz?

Eu já estou fazendo o setup do Detox, pode acompanhar o [andamento dessa task aqui](https://github.com/diego3g/rsxp-2023/issues/59)

Por conta de algumas coisas que discutimos nessa issue, [criamos uma votação em relação ao Expo Go e o Expo Dev Clients](https://github.com/diego3g/rsxp-2023/discussions/105)

Se poder dar o seu voto e opinião sobre o assunto lá eu ficaria muito feliz 😃
